crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -0,0 +1,253 @@
+        /// field will record the original lock expiration epoch, to be used when unstaking.
+        sui_token_lock: Option<EpochTimeLock>,
+        /// The pool tokens representing the amount of rewards the delegator can get back when they withdraw
+        /// from the pool. If this field is `none`, that means the delegation hasn't been activated yet.
+        pool_tokens: Option<Balance<DelegationToken>>,
+    }
+
+    /// Create a new, empty staking pool.
+    public(friend) fun new(validator_address: address, starting_epoch: u64) : StakingPool {
+        StakingPool {
+            validator_address,
+            starting_epoch,
+            sui_balance: 0,
+            rewards_pool: balance::zero(),
+            delegation_token_supply: balance::create_supply(DelegationToken {}),
+        }
+    }
+
+    /// Add rewards (in SUI) to the staking pool. 
+    /// Notice here we add SUI tokens but do not mint new pool tokens. Thus the exchange rate between SUI and pool tokens goes up.
+    public(friend) fun add_rewards(pool: &mut StakingPool, rewards: Balance<SUI>) {
+        pool.sui_balance = pool.sui_balance + balance::value(&rewards);
+        balance::join(&mut pool.rewards_pool, rewards);
+        // TODO: implement potential bonding period related bookkeeping
+    }
+
+    /// Request to delegate to a staking pool. The delegation doesn't get activated and the delegator doesn't get
+    /// the pool tokens until they activate the delegation on or after the `earliest_activation_epoch`.
+    public(friend) fun request_add_delegation(pool: &mut StakingPool, stake: Balance<SUI>, sui_token_lock: Option<EpochTimeLock>, ctx: &mut TxContext) {
+        assert!(balance::value(&stake) > 0, 0);
+        let delegation = Delegation {
+            id: object::new(ctx),
+            validator_address: pool.validator_address,
+            pool_starting_epoch: pool.starting_epoch,
+            earliest_activation_epoch: tx_context::epoch(ctx) + 1, // TODO: with bonding period this should be the epoch
+                                                                   // at which the bonding period ends
+            principal: stake,
+            sui_token_lock,
+            pool_tokens: option::none(),
+        };
+        transfer::transfer(delegation, tx_context::sender(ctx));
+    }
+
+    /// Activate a delegation. New pool tokens are minted at the current exchange rate and put into the
+    /// `pool_tokens` field of the delegation object.
+    /// After activation, the delegation officially counts toward the staking power of the validator.
+    /// Aborts if the pool mismatches, the delegation is already activated, or the delegation cannot be activated yet. 
+    public(friend) fun activate_delegation(pool: &mut StakingPool, delegation: &mut Delegation, ctx: &mut TxContext) {
+        assert!(
+            delegation.validator_address == pool.validator_address &&
+            delegation.pool_starting_epoch == pool.starting_epoch,
```

Comment:
Should it be "equal" or "less / more than"?

Reply:
Resolving this comment since it's longer relevant after the recent changes.